File: pegasusX/apps/warehouse-app-ios/WarehouseApp/Views/Dispatch/DispatchDriverList.swift
import SwiftUI

/// Reusable drivers segment for the Dispatch screen.
///
/// Renders available and unavailable driver cards with status badges,
/// VU capacity, and unavailability reasons.
struct DispatchDriverList: View {
    let availableDrivers: [AvailableDriver]
    let unavailableDrivers: [AvailableDriver]

    var body: some View {
        if availableDrivers.isEmpty && unavailableDrivers.isEmpty {
            ContentUnavailableView("No Drivers", systemImage: "person.badge.key", description: Text("No available drivers"))
        } else {
            ResponsiveGridContentWrapper {
                if !availableDrivers.isEmpty {
                    Section("Available") {
                        ForEach(availableDrivers) { driver in
                            driverRow(driver, unavailableDetail: false)
                        }
                    }
                }
                if !unavailableDrivers.isEmpty {
                    Section("Vehicle Unavailable") {
                        ForEach(unavailableDrivers) { driver in
                            driverRow(driver, unavailableDetail: true)
                        }
                    }
                }
            }
        }
    }

    @ViewBuilder
    private func driverRow(_ driver: AvailableDriver, unavailableDetail: Bool) -> some View {
        HStack(alignment: unavailableDetail ? .top : .center) {
            VStack(alignment: .leading, spacing: LabTheme.spacingXS) {
                Text(driver.name)
                    .font(.headline)
                Text(driverSubtitle(driver, unavailableDetail: unavailableDetail))
                    .font(.subheadline)
                    .foregroundStyle(.secondary)
                if unavailableDetail, let reason = driver.unavailableReason, !reason.isEmpty {
                    Text(vehicleUnavailableReasonLabel(reason))
                        .font(.caption)
                        .foregroundStyle(.orange)
                }
            }
            Spacer()
            VStack(alignment: .trailing, spacing: LabTheme.spacingXS) {
                WarehouseStatusBadge(text: driver.truckStatus.isEmpty ? "IDLE" : driver.truckStatus)
                if driver.maxVolumeVu > 0 {
                    Text("\(driver.maxVolumeVu, specifier: "%.0f") VU")
                        .font(.caption2)
                        .foregroundStyle(.secondary)
                }
            }
        }
    }

    private func driverSubtitle(_ driver: AvailableDriver, unavailableDetail: Bool) -> String {
        let vehicle = driver.vehicleLabel ?? ""
        let phone = driver.phone ?? ""
        let label = vehicle.isEmpty ? (phone.isEmpty ? (unavailableDetail ? "Assigned vehicle unavailable" : "No vehicle") : phone) : vehicle
        if let free = driver.freeVolumeVu, free > 0 {
            return "\(label) · \(String(format: "%.1f", free)) VU free"
        }
        return label
    }
}

File: pegasusX/apps/warehouse-app-android/app/src/main/java/com/pegasusx/warehouse/ui/components/DispatchDriverList.kt
package com.pegasusx.warehouse.ui.components

import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.grid.GridCells
import androidx.compose.foundation.lazy.grid.GridItemSpan
import androidx.compose.foundation.lazy.grid.LazyVerticalGrid
import androidx.compose.foundation.lazy.grid.items
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp
import com.pegasusx.warehouse.data.model.AvailableDriver
import com.pegasus.design.ui.PegasusStateKind
import com.pegasus.design.ui.PegasusStatePane
import com.pegasusx.warehouse.ui.screens.vehicles.vehicleUnavailableReasonLabel
import com.pegasusx.warehouse.ui.theme.PegasusSpacing

/**
 * Drivers segment of the Dispatch screen.
 *
 * Renders available and unavailable driver cards with status badges,
 * VU capacity, and unavailability reasons.
 */
@Composable
fun DispatchDriverList(
    availableDrivers: List<AvailableDriver>,
    unavailableDrivers: List<AvailableDriver>,
) {
    if (availableDrivers.isEmpty() && unavailableDrivers.isEmpty()) {
        PegasusStatePane(
            kind = PegasusStateKind.Empty,
            headline = "No drivers",
            body = "Available and unavailable drivers will appear here.",
        )
    } else {
        LazyVerticalGrid(
            columns = GridCells.Adaptive(minSize = 340.dp),
            contentPadding = PaddingValues(PegasusSpacing.lg),
            verticalArrangement = Arrangement.spacedBy(PegasusSpacing.md),
            horizontalArrangement = Arrangement.spacedBy(PegasusSpacing.md),
        ) {
            if (availableDrivers.isNotEmpty()) {
                item(span = { GridItemSpan(maxLineSpan) }) { WarehouseSectionTitle("Available") }
            }
            items(availableDrivers, key = { it.driverId }) { d ->
                val supporting = buildString {
                    append(d.vehicleLabel.ifBlank { d.phone.ifBlank { d.truckStatus.ifBlank { "No vehicle" } } })
                    if (d.freeVolumeVu != null && d.freeVolumeVu > 0) {
                        append(" · ${"%.1f".format(d.freeVolumeVu)} VU free")
                    }
                }
                WarehouseOpsListCard(
                    headline = d.name,
                    supporting = supporting,
                    status = d.truckStatus.ifBlank { "IDLE" },
                )
            }
            if (unavailableDrivers.isNotEmpty()) {
                item(span = { GridItemSpan(maxLineSpan) }) { WarehouseSectionTitle("Vehicle unavailable") }
            }
            items(unavailableDrivers, key = { "unavailable-${it.driverId}" }) { d ->
                WarehouseOpsListCard(
                    headline = d.name,
                    supporting = buildString {
                        append(d.vehicleLabel.ifBlank { d.phone.ifBlank { "Assigned vehicle unavailable" } })
                        if (!d.unavailableReason.isNullOrBlank()) {
                            append(" · ")
                            append(vehicleUnavailableReasonLabel(d.unavailableReason))
                        }
                    },
                    status = d.truckStatus.ifBlank { "UNAVAILABLE" },
                )
            }
        }
    }
}
